report/2.Analysis/ArduinoLanguagesCtrl.tex

```diff
@@ -135,18 +135,20 @@ \section{Existing languages}
 
 % Source: http://www.harbaum.org/till/nanovm/index.shtml
 \textbf{NanoVM} is a virtual machine (VM) for the Atmel AVR ATMega8 CPU.
-It allows programmers to write code in a subset of Java that can be run by Arduino and similar microcomputers.
+NanoVM allows programmers to write code in a subset of Java that can be run on the Arduino architecture and similar microcomputers.
 The VM includes many of the features that have made the Java VM (JVM) popular such as object-oriented programming, automatic dynamic memory allocation, and garbage collection.
 It comes packaged with native classes such as Object, System, and PrintStream.
 
 The following is a short description of the basic Java syntax. 
 Snippets \ref{lis:javaExample} and \ref{lis:javaPrinter} showcase the syntax.
 
-The program is started by calling the \texttt{main} function (snippet \ref{lis:javaExample}, line 5).
+The program begins by calling the \texttt{main} function (snippet \ref{lis:javaExample}, line 5).
 
-All statements are finished with a semi-colon.
+All statements are terminated with a semi-colon.
 
-In Java all variables and functions must be declared inside a class, and each class must be declared in a file for itself.
+In Java all variables and functions must be declared inside a class.
+
+In Java each separate file can only contain one public class, which must have the same name as the file itself. 
```

Comment:
dosent sound good when starting 2 sentences with "In Java"
- maybe start with 'Furthermore'